src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,219 @@
+  };
+
+  // #endregion
+  // #region reset
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setBody('');
+
+    setHasTitleError(false);
+    setHasUserIdError(false);
+    setBodyErrorMessage('');
+
+    onReset();
+  };
+
+  // #endregion
+  // #region submit
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!body) {
+      setBodyErrorMessage('Please enter some text');
+    } else if (body.length < 5) {
+      setBodyErrorMessage('Body should have at least 5 chars');
+    }
+
+    if (!title || !userId || body.length < 5) {
+      return;
+    }
+
+    onSubmit({
+      id: todo?.id || 0,
+      title,
+      body,
+      userId,
+      user: getUserById(userId),
+      completed: false,
+    });
+
+    reset();
+  };
+
+  // #endregion
+
+  return (
+    <form
+      action="/api/posts"
+      method="POST"
+      className="box"
+      onSubmit={handleSubmit}
+      onReset={reset}
+    >
+      <h2 className="title is-4">Create a todo</h2>
+
+      <div className="field">
+        <label className="label" htmlFor="todo-title">
+          Title
+        </label>
+
+        <div
+          className={classNames('control has-icons-left', {
+            'has-icons-right': hasTitleError,
+          })}
+        >
+          <input
+            id="todo-title"
+            ref={titleField}
+            className={classNames('input', {
+              'is-danger': hasTitleError,
+            })}
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter title"
+            value={title}
+            onChange={handleTitleChange}
+            onBlur={() => {
+              setHasTitleError(!title);
+            }}
+          />
+          <span className="icon is-small is-left">
+            <i className="fas fa-t"></i>
+          </span>
+
+          {hasTitleError && (
+            <span className="icon is-small is-right">
+              <i className="fas fa-exclamation-triangle has-text-danger"></i>
+            </span>
+          )}
+        </div>
+
+        {hasTitleError && (
+          <p className="help is-danger">Please enter a title</p>
+        )}
+      </div>
+
+      <div className="field">
+        <label className="label" htmlFor="todo-user-id">
+          Subject
+        </label>
+
+        <div className="control has-icons-left">
+          <div
+            className={classNames('select', {
+              'is-danger': hasUserIdError,
+            })}
+          >
+            <select
+              id="todo-user-id"
+              data-cy="userSelect"
+              value={userId}
+              onChange={handleUserIdChange}
+            >
+              <option value="0">Select a user</option>
+
+              {getAllUsers().map(user => (
+                <option value={user.id} key={user.id}>
+                  {user.name}
+                </option>
+              ))}
```

Comment:
Calling `getAllUsers()` directly in the JSX can lead to performance issues. Consider fetching the users once and storing them in a state or memoized value to avoid unnecessary computations during each render.